File: app/src/main/java/com/yolo/mvvmwanandroid/network/bean/Tag.kt
package com.yolo.mvvmwanandroid.network.bean

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize

/**
 * @author yolo.huang
 * @date 2020/8/20
 */
@Parcelize
@Entity
data class Tag(
    @PrimaryKey(autoGenerate = true)
    var id: Long,
    @ColumnInfo(name = "article_id")
    var articleId: Long,
    var name: String?,
    var url: String?
) : Parcelable

File: flutter_module/lib/bean/Tag.dart
class Tag {
  int _id;
  int _articleId;
  String _name;
  String _url;

  Tag(
      int id,
      int articleId,
      String name,
      String url
      ){
    this._id = id;
    this._articleId = articleId;
    this._name = name;
    this._url = url;
  }

  int get id=> _id;
  int get article=> _articleId;
  String get name=> _name;
  String get url => _url;

  Tag.fromJson(dynamic json) {
    _id = json["id"];
    _articleId = json["article"];
    _name = json["name"];
    _url = json["url"];
  }

  Map<String, dynamic> toJson() {
    var map = <String, dynamic>{};
    map["id"] = _id;
    map["article"] = _articleId;
    map["name"] = _name;
    map["url"] = _url;
    return map;
  }
}
